File: android/src/main/java/com/hitek/seamlessview/SeamlessView.java
package com.hitek.seamlessview;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.content.Context;
import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import android.widget.Toast;

public class SeamlessView extends WebView implements MethodCallHandler {
    private static final String CHANNEL = "com.hitek/seamlessview";
    private MethodChannel methodChannel;

    public SeamlessView(Context context, FlutterEngine flutterEngine) {
        super(context);
        methodChannel = new MethodChannel(flutterEngine.getDartExecutor().getBinaryMessenger(), CHANNEL);
        methodChannel.setMethodCallHandler(this);
        initWebViewSettings();
    }

    private void initWebViewSettings() {
        WebSettings settings = getSettings();
        settings.setJavaScriptEnabled(true); // Default setting, adjust as necessary
        settings.setDomStorageEnabled(true); // Enable DOM storage by default
    }

    @Override
    public void onMethodCall(MethodCall call, Result result) {
        switch (call.method) {
            case "seamlessInit":
                String url = call.argument("url");
                loadUrl(url);
                result.success(true);
                break;
            case "setUrl":
                String url2 = call.argument("url");
                loadUrl(url2);
                result.success(true);
                break;
            case "setJavaScriptEnabled":
                boolean javaScriptEnabled = call.argument("enabled");
                getSettings().setJavaScriptEnabled(javaScriptEnabled);
                result.success(true);
                break;
            case "setDomStorageEnabled":
                boolean domStorageEnabled = call.argument("enabled");
                getSettings().setDomStorageEnabled(domStorageEnabled);
                result.success(true);
                break;
            case "setDatabaseEnabled":
                boolean databaseEnabled = call.argument("enabled");
                getSettings().setDatabaseEnabled(databaseEnabled);
                result.success(true);
                break;
            case "setCacheMode":
                int cacheMode = call.argument("mode");
                getSettings().setCacheMode(cacheMode);
                result.success(true);
                break;
            case "setPageBackground":
                int color = call.argument("color");
                setBackgroundColor(color);
                result.success(true);
                break;
            case "setTextStyle":
                String textStyle = call.argument("style");
                WebSettings settings = getSettings();
                settings.setDefaultFontSize(16);
                settings.setBoldText(false);
                settings.setItalicText(false);
                if (textStyle.equals("bold")) {
                    settings.setBoldText(true);
                } else if (textStyle.equals("italic")) {
                    settings.setItalicText(true);
                } else if (textStyle.equals("normal")) {
                    // Do nothing, default font is normal
                } else if (textStyle.equals("strikethrough")) {
                    loadUrl("javascript:document.body.style.textDecoration='line-through';");
                } else if (textStyle.equals("underline")) {
                    loadUrl("javascript:document.body.style.textDecoration='underline';");
                } else {
                    // Handle unknown text style
                    result.error("unknown_text_style", "Unknown text style: " + textStyle, null);
                }
                result.success(true);
                break;
            case "setRequestInterceptor":
                WebViewClient requestInterceptor = new WebViewClient() {
                    @Override
                    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
                        String url = request.getUrl().toString();
                        if (url.startsWith("(link unavailable)")) {
                            // Handle requests to (link unavailable)
                            return true;
                        } else if (url.startsWith("http://")) {
                            // Allow HTTP requests to proceed
                            return false;
                        } else if (url.startsWith("https://")) {
                            // Allow HTTPS requests to proceed
                            return false;
                        } else {
                            // Block unknown requests
                            return true;
                        }
                    }
                };
                setWebViewClient(requestInterceptor);
                result.success(true);
                break;
            case "setErrorHandler":
                WebViewClient errorHandler = new WebViewClient() {
                    @Override
                    public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
                        // Implement error handler logic here
                        Toast.makeText(getContext(), "Error loading page: " + description, Toast.LENGTH_SHORT).show();
                        // Handle error further if needed
                    }
                };
                setWebViewClient(errorHandler);
                result.success(true);
                break;
            case "setUserAgent":
                String userAgent = call.argument("userAgent");
                getSettings().setUserAgentString(userAgent);
                result.success(true);
                break;
            case "setSupportZoom":
                boolean supportZoom = call.argument("supportZoom");
                getSettings().setSupportZoom(supportZoom);
                result.success(true);
                break;
            case "setBuiltInZoomControls":
                boolean builtInZoomControls = call.argument("builtInZoomControls");
                getSettings().setBuiltInZoomControls(builtInZoomControls);
                result.success(true);
                break;
            case "setDisplayZoomControls":
                boolean displayZoomControls = call.argument("displayZoomControls");
                getSettings().setDisplayZoomControls(displayZoomControls);
                result.success(true);
                break;
            case "setLoadWithOverviewMode":
                boolean loadWithOverviewMode = call.argument("loadWithOverviewMode");
                getSettings().setLoadWithOverviewMode(loadWithOverviewMode);
                result.success(true);
                break;
            case "setUseWideViewPort":
                boolean useWideViewPort = call.argument("useWideViewPort");
                getSettings().setUseWideViewPort(useWideViewPort);
                result.success(true);
                break;
            case "reload":
                reload();
                result.success(true);
                break;
            case "showUrlBar":
                boolean showUrlBar = call.argument("showUrlBar");
                if (showUrlBar) {
                    // Show the URL bar
                    ViewGroup.LayoutParams layoutParams = getLayoutParams();
                    layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT;
                    setLayoutParams(layoutParams);
                } else {
                    // Hide the URL bar
                    ViewGroup.LayoutParams layoutParams = getLayoutParams();
                    layoutParams.height = 0;
                    setLayoutParams(layoutParams);
                }
                result.success(true);
                break;
            default:
                result.notImplemented();
                break;
        }
    }

    @Override
    public void onMethodCallWithContext(MethodCall call, Result result, MethodCallHandlerContext context) {
        onMethodCall(call, result);
    }
}

File: ios/Classes/SeamlessView.swift
import UIKit
import WebKit

class SeamlessView: WKWebView, WKNavigationDelegate {

    static let CHANNEL = "com.hitek/seamlessview"

    var methodChannel: FlutterMethodChannel

    init(frame: CGRect, configuration: WKWebViewConfiguration, flutterEngine: FlutterEngine) {
        methodChannel = FlutterMethodChannel(name: SeamlessView.CHANNEL, binaryMessenger: flutterEngine.binaryMessenger)
        methodChannel.setMethodCallHandler({ [weak self] (call, result) in
            self?.handleMethodCall(call, result)
        })
        super.init(frame: frame, configuration: configuration)
        navigationDelegate = self
    }

    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    func handleMethodCall(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        switch call.method {
        case "setUrl":
            if let url = call.arguments as? String {
                load(URLRequest(url: URL(string: url)!))
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setJavaScriptEnabled":
            if let enabled = call.arguments as? Bool {
                configuration.preferences.javaScriptEnabled = enabled
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setDomStorageEnabled":
            if let enabled = call.arguments as? Bool {
                configuration.preferences.domStorageEnabled = enabled
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setDatabaseEnabled":
            if let enabled = call.arguments as? Bool {
                configuration.preferences.databaseEnabled = enabled
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setCacheMode":
            if let mode = call.arguments as? Int {
                configuration.preferences.cacheModel = WKCacheModel(rawValue: mode)!
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setPageBackground":
            if let color = call.arguments as? UIColor {
                backgroundColor = color
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setTextStyle":
            if let textStyle = call.arguments as? String {
                switch textStyle {
                case "bold":
                    let font = webView.font
                    let boldFont = UIFont.boldSystemFont(ofSize: font.pointSize)
                    webView.font = boldFont
                case "italic":
                    let font = webView.font
                    let italicFont = UIFont.italicSystemFont(ofSize: font.pointSize)
                    webView.font = italicFont
                case "normal":
                    let font = webView.font
                    let normalFont = UIFont.systemFont(ofSize: font.pointSize)
                    webView.font = normalFont
                case "strikethrough":
                    webView.stringByEvaluatingJavaScript(from: "document.body.style.textDecoration = 'line-through';")
                case "underline":
                    webView.stringByEvaluatingJavaScript(from: "document.body.style.textDecoration = 'underline';")
                default:
                    result(FlutterMethodNotImplemented)
                    return
                }
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setRequestInterceptor":
            if let urlPatterns = call.arguments as? [String] {
                let requestInterceptor = RequestInterceptor(urlPatterns: urlPatterns)
                webView.uiDelegate = requestInterceptor
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        case "setErrorHandler":
            if let errorHandler = call.arguments as? (String, String) -> Void {
                self.errorHandler = errorHandler
                result(true)
            } else {
                result(FlutterMethodNotImplemented)
            }
        default:
            result(FlutterMethodNotImplemented)
        }
    }
}